### PR TITLE
feature: Add `vendors` webpack for faster development builds.

### PR DESCRIPTION
**Pull request by jmahc**
I added a `postinstall` script to the `package.json` file that will run the new `dev` script installing vendor files and placing a `vendor-manifest.json` file and `vendor-dll.js` file in the `body` of the `index.html` file.
This results in faster `development` compilation times, as webpack has already prepared the `dependencies` found in the `package.json` file.

### CODE REVIEW

File: webpack.config.ts [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
 import chalk from 'chalk'
+import fs from 'fs'
+import HtmlWebpackPlugin from 'html-webpack-plugin'
+import path from 'path'
+import webpack from 'webpack'
+
+const currentDir = process.cwd()
+const appDirectory = fs.realpathSync(currentDir)
+const resolvePath = (relativePath: string) =>
+  path.resolve(appDirectory, relativePath)
```
Comment (zoomercode) [inline webpack.config.ts]: **cringe** Y no `require.resolve()` ?

**Comment by zoomercode** [comment]
very nice, im takin a look

File: webpack.config.ts [hunk 2]
```diff
@@ -40,28 +79,7 @@ const config: webpack.Configuration = {
...
-              [
-                '@babel/preset-env',
-                {
-                  targets: {
-                    chrome: 80,
-                  },
-                },
-              ],
-            ],
+            configFile: resolvePath('babel.config.js'),
```
Comment (zoomercode) [inline webpack.config.ts]: why not just `require('./babel.config.js')` ?

Comment (zoomercode) [inline webpack.config.ts]: should just find `babel.config.js` without this right?

Comment (jmahc) [reply webpack.config.ts]: I do not know off the top of my head, try commenting it out?

**Comment by zoomercode** [comment]
not sure I like the `postinstall` hook, since when I install a dev dep it still runs. When I install "mocha" for example it runs and adds an extra 16s to the command

File: package.json [hunk 3]
```diff
@@ -3,10 +3,10 @@
   "version": "1.0.0",
   "description": "",
   "scripts": {
-    "build": "NODE_ENV=production webpack",
+    "build": "cross-env NODE_ENV=production webpack",
```
Comment (zoomercode) [inline package.json]: It's all fun and games until windows users show up

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,5 @@
 node_modules
 dist
 ws-proxy
+.env
+dll
```
Comment (zoomercode) [inline .gitignore]: I thought editorconfig would ensure EOF newline?

**Comment by zoomerdev** [comment]
image

**Pull request merged**